x/mint/keeper/keeper.go

```diff
@@ -315,3 +344,96 @@ func (k Keeper) createDeveloperVestingModuleAccount(ctx sdk.Context, amount sdk.
+// and adding it to any left overs from the previous epoch. The left overs might be stemming from the inability
+// to distribute decimal truncations less than 1. As a result, we store them in the store until the next epoch.
+// These truncation distributions have eventual guarantees. That is, they are guaranteed to be distributed
+// eventually but not necessarily during the same epoch.
+// Returns an error if the module account name is any other than mint or developer vesting is given.
+// The truncation delta is calculated by subtracting amountDistributed from provisions and adding to
+// any leftover truncations from the previous epoch.
+// Therefore, provisions must be greater than or equal to the amount distributed. Errors if not.
+// For any amount to be distributed from the mint module account, it mints the estimated truncation amount
+// before distributing it to the community pool.
+// Additionally, it errors in the following cases:
+// - unable to mint tokens
+// - unable to fund the community pool
+// nolint: unused
+func (k Keeper) handleTruncationDelta(ctx sdk.Context, moduleAccountName string, provisions sdk.DecCoin, amountDistributed sdk.Int) (sdk.Int, error) {
+	if moduleAccountName != types.DeveloperVestingModuleAcctName && moduleAccountName != types.ModuleName {
+		return sdk.Int{}, sdkerrors.Wrapf(types.ErrInvalidModuleAccountGiven, "truncation delta can only be handled by (%s) or (%s) module accounts but (%s) was given", types.DeveloperVestingModuleAcctName, types.ModuleName, moduleAccountName)
+	}
+	if provisions.Amount.LT(amountDistributed.ToDec()) {
+		return sdk.Int{}, sdkerrors.Wrapf(types.ErrInvalidAmount, "provisions (%s) must be greater than or equal to amount disributed (%s)", provisions.Amount, amountDistributed)
+	}
+
+	deltaAmount, err := k.calculateTotalTruncationDelta(ctx, moduleAccountName, provisions.Amount, amountDistributed)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	if deltaAmount.LT(sdk.OneDec()) {
+		if err := k.SetTruncationDelta(ctx, moduleAccountName, deltaAmount); err != nil {
+			return sdk.Int{}, err
+		}
+		return sdk.ZeroInt(), nil
+	}
+
+	// N.B: Truncation is acceptable because we check delta at the end of every epoch.
+	// As a result, actual minted distributions always approach the expected value.
+	truncationDeltaToDistribute := deltaAmount.TruncateInt()
+	// For funding from mint module account, we must pre-mint first.
+	if moduleAccountName == types.ModuleName {
+		if err := k.mintCoins(ctx, sdk.NewCoins(sdk.NewCoin(provisions.Denom, truncationDeltaToDistribute))); err != nil {
+			return sdk.Int{}, err
+		}
+	}
+	if err := k.communityPoolKeeper.FundCommunityPool(ctx, sdk.NewCoins(sdk.NewCoin(provisions.Denom, truncationDeltaToDistribute)), k.accountKeeper.GetModuleAddress(moduleAccountName)); err != nil {
+		return sdk.Int{}, err
+	}
+
+	newDelta := deltaAmount.Sub(truncationDeltaToDistribute.ToDec())
```

Comment:
I think this would be a good place to guarantee a delta that is less that 1 but greater than 0 right? That or within the `SetTruncationDelta` function itself.

Reply:
Already added in `SetTruncationDelta`

Reply:
Nvm, that was for 0, adding now

Reply:
added